src/engine.rs

```diff
@@ -1,19 +1,53 @@
 use borsh::{BorshDeserialize, BorshSerialize};
 use evm::backend::{Apply, ApplyBackend, Backend, Basic, Log};
-use evm::executor::{MemoryStackState, StackExecutor, StackSubstateMetadata};
+use evm::executor::{StackExecutor, StackSubstateMetadata};
 use evm::ExitFatal;
 use evm::{Config, CreateScheme, ExitError, ExitReason};
 
 use crate::connector::EthConnectorContract;
-use crate::map::LookupMap;
-use crate::parameters::{FunctionCallArgs, NewCallArgs, SubmitResult, ViewCallArgs};
+#[cfg(feature = "contract")]
+use crate::contract::current_address;
+use crate::map::{BijectionMap, LookupMap};
+use crate::parameters::{
+    FunctionCallArgs, NEP141FtOnTransferArgs, NewCallArgs, PromiseCreateArgs, SubmitResult,
+    ViewCallArgs,
+};
 
 use crate::precompiles;
-use crate::prelude::{Address, TryInto, Vec, H256, U256};
+use crate::prelude::{Address, ToString, TryInto, Vec, H256, U256};
 use crate::sdk;
+use crate::state::AuroraStackState;
 use crate::storage::{address_to_key, bytes_to_key, storage_to_key, KeyPrefix, KeyPrefixU8};
 use crate::types::{u256_to_arr, AccountId, Wei};
 
+#[cfg(not(feature = "contract"))]
+pub fn current_address() -> Address {
+    crate::types::near_account_to_evm_address("engine".as_bytes())
+}
+
+macro_rules! unwrap_res_or_finish {
+    ($e:expr, $output:expr) => {
+        match $e {
+            Ok(v) => v,
+            Err(_e) => {
+                #[cfg(feature = "log")]
+                sdk::log(crate::prelude::format!("{:?}", _e).as_str());
+                sdk::return_output($output);
+                return;
+            }
+        }
+    };
+}
+
+macro_rules! assert_or_finish {
```

Comment:
Just a nit, these macro names feel a bit off. Technically just `unwrap` conveys that meaning that the app will unwrap, or panic. Also, don't we have these already? `sdk_unwrap` etc.?

Reply:
See also previous discussion: https://github.com/aurora-is-near/aurora-engine/pull/51#discussion_r637229881

Essentially my opinion is that we should be able to remove these macros altogether, but we won't do it now since apparently it's non-trivial and this PR is important.

Reply:
Notice that we are not panicking, so we can't use `sdk_unwrap` there. For the name, tbh it is clear to me, we are making emphasis that if unwrap is not possible we will be finishing. 

It is very similar to using `?`. It wasn't done this way, since we should've returned an object of type `Result<Vec<u8>, Vec<u8>>` which is more inefficient, and doesn't look like a nice API.

Reply:
Sure, not critical for this PR anyways.